apps/pxweb2/src/app/context/TableDataProvider.tsx

```diff
@@ -65,9 +190,485 @@ const TableDataProvider: React.FC<TableDataProviderProps> = ({ children }) => {
+              const newValue = structuredClone(value);
+
+              // Values are assumed to be sorted in the right order in the VariablesProvider
+              // Find the index where the new value should be inserted
+              const valueIndex = selection?.valueCodes?.indexOf(value.code);
+              if (valueIndex !== undefined && valueIndex !== -1) {
+                existingVariable?.values.splice(valueIndex, 0, newValue);
+              } else {
+                existingVariable?.values.push(newValue);
+              }
+            }
+          });
+        }
+      }
+
+      // --- Merge data ---
+
+      // Dimensions in accumulatedData
+      const dimensions: string[] = [];
+
+      // Variable order in accumulatedData can be different from the variable order in the new API-call.
+      // Because of this we need to create a mapping of the variable order in accumulatedData to the variable order in the new API-call.
+      const dimensionsMap: number[] = getDimensionsMap(
+        accumulatedData.metadata,
+        pxTable.metadata
+      );
+
+      // Update the accumulated data cube with the new data
+      updateCube(accumulatedData, pxTable, dimensions, 0, dimensionsMap);
+    }
+  }
+
+  /**
+   * Get dimension mapper for how variables are located in new data vs in accumulated data
+   *
+   * @param accumulatedMetadata - PxTableMetadata object for the accumulated data.
+   * @param newMetadata - PxTableMetadata object for the new data.
+   * @returns An array with the index of the variables in the new data.
+   */
+  function getDimensionsMap(
+    accumulatedMetadata: PxTableMetadata,
+    newMetadata: PxTableMetadata
+  ): number[] {
+    const dimensionsMap: number[] = [];
+
+    for (let i = 0; i < accumulatedMetadata.variables.length; i++) {
+      const index = newMetadata.variables.findIndex(
+        (variable) => variable.id === accumulatedMetadata.variables[i].id
+      );
+      dimensionsMap[i] = index;
+    }
+
+    return dimensionsMap;
+  }
+
+  /**
+   * Updates the accumulated data cube (`accData`) with new data from `newData` based on the specified dimensions.
+   *
+   * @param accData - The accumulated data cube to be updated.
+   * @param newData - The new data cube containing the updated values.
+   * @param dimensions - An array of dimension values used to navigate through the data cubes.
+   * @param dimensionIndex - The current index of the dimension being processed.
+   * @param dimensionsMap - A mapping of dimension indices to their corresponding positions in the dimensions array.
+   *
+   * This function recursively navigates through the dimensions of the data cubes. For each dimension, it finds the corresponding variable in the new data cube and updates the accumulated data cube with the new values. If the current dimension is the last one, it retrieves the data value from the new data cube and sets it in the accumulated data cube. Otherwise, it continues to the next dimension.
```

Comment:
Maybe break this comment into several lines, so we do not have to scroll to the right to read it?

Reply:
I break this up also.